### PR TITLE
Add Matrix3x2 transform methods

### PR DESCRIPTION
**Pull request by JeremyKuhne**
This implements #47940 for System.Drawing.Common.

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
@safern Any idea on how to tackle the build errors here? Getting a lot of this sort of thing:

```
src\libraries\System.IO.Ports\src\System.IO.Ports.csproj(0,0): error NU1201: (NETCORE_ENGINEERING_TELEMETRY=Restore) Project System.Numerics.Vectors is not compatible with netstandard2.0 (.NETStandard,Version=v2.0). Project System.Numerics.Vectors supports: net6.0 (.NETCoreApp,Version=v6.0)
```

File: src/libraries/System.Drawing.Common/src/System/Drawing/Drawing2D/Matrix.cs [hunk 1]
```diff
@@ -93,6 +116,30 @@ public float[] Elements
...
+        {
+            get
+            {
+                Matrix3x2 matrix = default;
+                Gdip.CheckStatus(Gdip.GdipGetMatrixElements(new HandleRef(this, NativeMatrix), (float*)&matrix));
+                return matrix;
+            }
+            set
+            {
+                Gdip.CheckStatus(Gdip.GdipSetMatrixElements(
```
Comment (stephentoub) [inline src/libraries/System.Drawing.Common/src/System/Drawing/Drawing2D/Matrix.cs:132]: So GDI exposes a GetMatrixElements that takes a single pointer, but a SetMatrixElements that takes each float individually?

File: src/libraries/System.Drawing.Common/src/System/Drawing/Graphics.cs [hunk 2]
```diff
@@ -332,6 +332,53 @@ public Matrix Transform
...
+        /// <remarks>
+        /// This is a more performant alternative to <see cref="Transform"/> that does not need disposal.
+        /// </remarks>
+        public unsafe Matrix3x2 TransformElements
+        {
+            get
+            {
+                Gdip.CheckStatus(Gdip.GdipCreateMatrix(out IntPtr nativeMatrix));
+                Gdip.CheckStatus(Gdip.GdipGetWorldTransform(
+                    new HandleRef(this, NativeGraphics), new HandleRef(null, nativeMatrix)));
```
Comment (stephentoub) [inline src/libraries/System.Drawing.Common/src/System/Drawing/Graphics.cs:347]: This doesn't need to be inside the try?  Seems like if this fails, we won't clean up nativeMatrix?

File: src/libraries/System.Drawing.Common/src/System/Drawing/Drawing2D/Matrix.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/libraries/System.Drawing.Common/src/System/Drawing/Drawing2D/Matrix.cs:132]: Just triple checked. While `GpMatrix` has a constructor that takes an array of points, it isn't projected into the flat api or the public `Matrix` class. 😞 I could create the identity matrix and set all the points at once, but I presume that would be much slower.

File: src/libraries/System.Drawing.Common/src/System/Drawing/Graphics.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/libraries/System.Drawing.Common/src/System/Drawing/Graphics.cs:347]: You're right, thanks!

**Review by tannergooding — APPROVED** [review]

**Pull request merged**